### DISCUSSION TITLE
Toast.Trigger component for uncontrolled orchestration?

### DISCUSSION THREAD

**Discussion by afzalsayed96** [radix-ui/primitives#2907] [Help]
Having used other radix-ui primitives like dialog and popover, I feel a bit lost about using Toast in an uncontrolled manner.

Why doesn't `Toast` have a `Trigger` component similar to `Dialog`, `Popover` and other primitives?

**Comment by jjenzz** [comment]
@afzalsayed96 toasts are usually triggered in response to a side effect that has occurred in your application, whereas `Dialog` should generally always open when `Trigger` is pressed.

for instance you might do something like this:

```tsx
const [isToastOpen, setIsToastOpen] = React.useState(false);

const handleButtonClick = () => {
  if (someSideEffectSucceeded) setIsToastOpen(true);
};

return (
  <div>
    <button onClick={handleButtonClick}>Go</button>
    <Toast.Root open={isToastOpen} onOpenChange={setIsToastOpen}>{/* ... */}</Toast.Root>
  </div>
);
```

the `someSideEffectSucceeded` part is important, Radix doesn't know what that is in order to trigger the toast for you—it could even be something asynchronous.
